otj-metrics-core/src/main/java/com/opentable/metrics/OTInstrumentedHandler.java

```diff
@@ -0,0 +1,343 @@
+ */
+@SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+public class OTInstrumentedHandler extends HandlerWrapper {
+    private final MetricRegistry metricRegistry;
+
+    private String name;
+    private final String prefix;
+
+    // the requests handled by this handler, excluding active
+    private Timer requests;
+
+    // the number of dispatches seen by this handler, excluding active
+    private Timer dispatches;
+
+    // the number of active requests
+    private Counter activeRequests;
+
+    // the number of active dispatches
+    private Counter activeDispatches;
+
+    // the number of requests currently suspended.
+    private Counter activeSuspended;
+
+    // the number of requests that have been asynchronously dispatched
+    private Meter asyncDispatches;
+
+    // the number of requests that expired while suspended
+    private Meter asyncTimeouts;
+
+    private Meter[] responses;
+
+    private Timer getRequests;
+    private Timer postRequests;
+    private Timer headRequests;
+    private Timer putRequests;
+    private Timer deleteRequests;
+    private Timer optionsRequests;
+    private Timer traceRequests;
+    private Timer connectRequests;
+    private Timer moveRequests;
+    private Timer otherRequests;
+
+    private AsyncListener listener;
+
+    /**
+     * Create a new instrumented handler using a given metrics registry.
+     *
+     * @param registry the registry for the metrics
+     */
+    public OTInstrumentedHandler(MetricRegistry registry) {
+        this(registry, null);
+    }
+
+    /**
+     * Create a new instrumented handler using a given metrics registry.
+     *
+     * @param registry the registry for the metrics
+     * @param prefix   the prefix to use for the metrics names
+     */
+    public OTInstrumentedHandler(MetricRegistry registry, String prefix) {
+        this.metricRegistry = registry;
+        this.prefix = prefix;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        final String prefix = this.prefix == null ? name(getHandler().getClass(), name) : name(this.prefix, name);
+
+        this.requests = metricRegistry.timer(name(prefix, "requests"));
+        this.dispatches = metricRegistry.timer(name(prefix, "dispatches"));
+
+        this.activeRequests = metricRegistry.counter(name(prefix, "active-requests"));
+        this.activeDispatches = metricRegistry.counter(name(prefix, "active-dispatches"));
+        this.activeSuspended = metricRegistry.counter(name(prefix, "active-suspended"));
+
+        this.asyncDispatches = metricRegistry.meter(name(prefix, "async-dispatches"));
+        this.asyncTimeouts = metricRegistry.meter(name(prefix, "async-timeouts"));
+
+        this.responses = new Meter[]{
```

Comment:
I would have rather built a HashMap of something like [Response Status Family](https://docs.oracle.com/javaee/7/api/javax/ws/rs/core/Response.Status.Family.html) to meter. That would make the code easier to read, as you wouldn't have to manipulate array index as is done in `updateResponses`.

Reply:
This was taken directly from DropWizard Metrics. Let me post a diff to show what was changed.